### ISSUE TITLE
okta_app_oauth Resource Fails on Apply with 400 if refresh_token_leeway is either 0 or not present in .tf

### ISSUE THREAD

**Issue by emanor-okta** [okta/terraform-provider-okta#1584]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.4.6
on darwin_amd64
+ provider registry.terraform.io/okta/okta v4.0.0

### Affected Resource(s)

- okta_app_oauth

### Terraform Configuration Files

```hcl
resource "okta_app_oauth" "my_applicationName" {
  auto_key_rotation = true
  consent_method = "REQUIRED"
  grant_types = [
    "authorization_code",
    "refresh_token"
  ]
  hide_ios = true
  hide_web = true
  implicit_assignment = false
  issuer_mode = "DYNAMIC"
  label = "RSC"
  login_mode = "DISABLED"
  pkce_required = true 
  post_logout_redirect_uris = [
    "http://127.0.0.1:2121/logout"
  ]
  redirect_uris = [
    "com.oktapreview.domain:/callback"
  ]
  refresh_token_rotation = "ROTATE"
  refresh_token_leeway = 0 #or not present since it is marked as optional in docss
  response_types = [
    "code"
  ]
  token_endpoint_auth_method = "none"
  type = "native"
}
```

### Debug Output

https://gist.github.com/emanor-okta/0c44ad11bda95f23b321db6a3b7e7741

### Expected Behavior

For the app to be added to the Org when `refresh_token_leeway` is either not included, or it's value is set to `0`.

#### Can this be done in the Admin UI?

No

#### Can this be done in the actual API call?

Yes

### Actual Behavior

If `refresh_token_leeway` is set to `0` or not present, then during the `apply` phase the provider will set it's value to `null`.
```
...
   "refresh_token": {
    "leeway": null,
    "rotation_type": "ROTATE"
   },
...
```

Note when the plan phase is run and Terraform displays what will change correctly.
When set to `0` in resource
```
...
      + refresh_token_leeway       = 0
      + refresh_token_rotation     = "ROTATE"
...
```
When not included in resource
```
...
      + refresh_token_leeway       = (known after apply)
      + refresh_token_rotation     = "ROTATE"
...
```

### Steps to Reproduce

1. `terraform apply`

### References

- https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_oauth#refresh_token_leeway

**Comment by duytiennguyen-okta** [comment]
OKTA internal reference https://oktainc.atlassian.net/browse/OKTA-618781

**Comment by duytiennguyen-okta** [comment]
This is fixed in the release [4.0.2](https://github.com/okta/terraform-provider-okta/releases/tag/v4.0.2) from https://github.com/okta/terraform-provider-okta/pull/1596
